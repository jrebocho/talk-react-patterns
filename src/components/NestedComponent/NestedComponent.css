.outer-box {
  width: 280px;
  height: 180px;
  padding: 20px;
  text-align: center;
}

.box:hover {
  cursor: pointer;
}

.green {
  background-color: green;
  color: yellow;
}

.red {
  background-color: red;
  color: white;
}

.yellow {
  background-color: yellow;
  padding: 40px;
}