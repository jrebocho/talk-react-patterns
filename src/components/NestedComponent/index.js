import React, { Component } from 'react'
import './NestedComponent.css'
import Stateless from '../Stateless'

class NestedComponent extends Component {
  constructor(props) {
    super(props)
    this.state = { on: false }
    this.toggle = this.toggle.bind(this)
  }

  toggle() {
    this.setState(prevState => ({
      on: !prevState.on
    }))
  }

  render() {
    const boxColor = this.state.on ? 'green' : 'red'

    return (
      <div className={`outer-box ${boxColor}`} onClick={this.toggle}>
        <YellowBox on={this.state.on} />
      </div>
    )
  }
}

const YellowBox = ({ on }) => (
  <div className="yellow">
    <Stateless on={on} />
  </div>
)

export default NestedComponent
