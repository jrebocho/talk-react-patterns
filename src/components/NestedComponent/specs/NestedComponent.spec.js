import React from 'react'
import { shallow } from 'enzyme'
import NestedComponent from '../../NestedComponent'

describe('<NestedComponent />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<NestedComponent />)
    expect(wrapper).toMatchSnapshot()
  })

  describe('when the component is clicked', () => {
    it('renders correctly', () => {
      const wrapper = shallow(<NestedComponent />)
      wrapper.simulate('click')

      expect(wrapper).toMatchSnapshot()
    })
  })
})
