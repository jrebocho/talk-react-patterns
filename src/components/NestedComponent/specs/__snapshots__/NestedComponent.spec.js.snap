// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NestedComponent /> renders correctly 1`] = `
<div
  className="outer-box red"
  onClick={[Function]}
>
  <YellowBox
    on={false}
  />
</div>
`;

exports[`<NestedComponent /> when the component is clicked renders correctly 1`] = `
<div
  className="outer-box green"
  onClick={[Function]}
>
  <YellowBox
    on={true}
  />
</div>
`;
